test_runner/performance/benchbase_tpc_c_helpers/upload_results_to_perf_test_results.py

```diff
@@ -0,0 +1,587 @@
+            "Median Latency (microseconds)": "Median Latency",
+            "Minimum Latency (microseconds)": "Minimum Latency",
+            "25th Percentile Latency (microseconds)": "25th Percentile Latency",
+            "90th Percentile Latency (microseconds)": "90th Percentile Latency",
+            "99th Percentile Latency (microseconds)": "99th Percentile Latency",
+            "75th Percentile Latency (microseconds)": "75th Percentile Latency",
+            "Average Latency (microseconds)": "Average Latency",
+        }
+
+        for json_key, clean_name in latency_metrics.items():
+            if json_key in latency_data:
+                metrics.append((clean_name, latency_data[json_key]))
+
+    return metrics
+
+
+def build_labels(summary_data, project_id):
+    """Build labels JSON object from summary data and project info."""
+    labels = {}
+
+    # Extract required label keys from summary data
+    label_keys = [
+        "DBMS Type",
+        "DBMS Version",
+        "Benchmark Type",
+        "Final State",
+        "isolation",
+        "scalefactor",
+        "terminals",
+    ]
+
+    for key in label_keys:
+        if key in summary_data:
+            labels[key] = summary_data[key]
+
+    # Add project_id from workflow
+    labels["project_id"] = project_id
+
+    return labels
+
+
+def build_suit_name(scalefactor, terminals, run_type, min_cu, max_cu):
+    """Build the suit name according to specification."""
+    return f"benchbase-tpc-c-{scalefactor}-{terminals}-{run_type}-{min_cu}-{max_cu}"
+
+
+def convert_timestamp_to_utc(timestamp_ms):
+    """Convert millisecond timestamp to PostgreSQL-compatible UTC timestamp."""
+    try:
+        dt = datetime.fromtimestamp(timestamp_ms / 1000.0, tz=timezone.utc)
+        return dt.isoformat()
+    except (ValueError, TypeError) as e:
+        print(f"Warning: Could not convert timestamp {timestamp_ms}: {e}")
+        return datetime.now(timezone.utc).isoformat()
+
+
+def insert_metrics(conn, metrics_data):
+    """Insert metrics data into the perf_test_results table."""
+    insert_query = """
+    INSERT INTO perf_test_results
+    (suit, revision, platform, metric_name, metric_value, metric_unit,
+     metric_report_type, recorded_at_timestamp, labels)
+    VALUES (%(suit)s, %(revision)s, %(platform)s, %(metric_name)s, %(metric_value)s,
+            %(metric_unit)s, %(metric_report_type)s, %(recorded_at_timestamp)s, %(labels)s)
+    """
+
+    try:
+        with conn.cursor() as cursor:
+            cursor.executemany(insert_query, metrics_data)
+        conn.commit()
+        print(f"Successfully inserted {len(metrics_data)} metrics into perf_test_results")
+
+        # Log some sample data for verification
+        if metrics_data:
+            print(
+                f"Sample metric: {metrics_data[0]['metric_name']} = {metrics_data[0]['metric_value']} {metrics_data[0]['metric_unit']}"
+            )
+
+    except Exception as e:
+        conn.rollback()
```

Comment:
Probably, rollback() adds nothing here, as the transaction will be aborted anyway on the client exit.